src/main/java/javaslang/control/Try.java

```diff
@@ -234,6 +235,35 @@
         return flatMap(ignored -> Try.run(runnable));
     }
 
+    /**
+     * Runs the given checked function if this is a {@code Success},
+     * passing the result of the current expression to it.
+     * If this expression is a {@code Failure} then it'll return a new
+     * {@code Failure} of type R with the original exception.
+     *
+     * The main use case is chaining checked functions using method references:
+     *
+     * <pre>
+     * <code>
+     * Try.of(() -&gt; 100)
+     *    .andThen(x -&gt; x + 100)
+     *    .andThen(x -&gt; x * 20);
+     *
+     * </code>
+     * </pre>
+     *
+     * @param f A checked function taking a single argument.
+     * @return a new {@code Try}
+     */
+    default <R> Try<R> andThen(CheckedFunction1<T, R> f) {
+
+        if (this.isFailure()) {
+            return new Failure<>(this.failed().get());
+        }
+
+        return Try.of(() -> f.apply(this.get()));
```

Comment:
Please replace the method with:

```java
    default <R> Try<R> andThen(CheckedFunction1<T, R> f) {
        return flatMap(value -> Try.of(() -> f.apply(value)));
    }
```

Scala folks would say _flatMap that shit_ :-)